synapseclient/core/version_check.py

```diff
@@ -15,84 +15,176 @@
 import json
 import re
 import sys
+import urllib.request
+from typing import Optional
 
 import requests
 
 import synapseclient
 
 _VERSION_URL = "https://raw.githubusercontent.com/Sage-Bionetworks/synapsePythonClient/master/synapseclient/synapsePythonClient"  # noqa
+_PYPI_JSON_URL = "https://pypi.org/pypi/synapseclient/json"
+_RELEASE_NOTES_URL = "https://python-docs.synapse.org/news/"
 
 
 def version_check(
```

Comment:
When `version_check` is being called from within the client can you swap this logic over to use the `logger` to print out info level messages instead of `print` and `sys.xxx.write` messages?

https://github.com/Sage-Bionetworks/synapsePythonClient/blob/8663e459b321270b65a0305c5126b0e955648fc9/synapseclient/core/utils.py#L1044-L1049 is similar logic I've added elsewhere so that we can maintain backwards compatibility in the event a user is calling this function.

Reply:
yep!

Reply:
Done